vmware/ngc/NGC-Plugin/src/main/java/org/opensds/vmware/ngc/dao/impl/DeviceRepositoryImpl.java

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package org.opensds.vmware.ngc.dao.impl;
+
+import org.opensds.vmware.ngc.common.Storage;
+import org.opensds.vmware.ngc.common.StorageFactory;
+import org.opensds.vmware.ngc.models.StorageMO;
+import org.opensds.vmware.ngc.dao.DeviceRepository;
+import org.opensds.vmware.ngc.model.DeviceInfo;
+import org.opensds.vmware.ngc.util.Constant;
+import org.opensds.vmware.ngc.util.FileManager;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.io.*;
+import java.util.Map;
+import java.util.concurrent.*;
+
+@Repository
+@Lazy(value = false)
+public class DeviceRepositoryImpl implements DeviceRepository {
+
+    private static final Log logger = LogFactory.getLog(DeviceRepositoryImpl.class);
+    private static Map<String, DeviceInfo> CACHE_DEVICE_INFO = new ConcurrentHashMap<>();
+    private static Map<String, Storage> LOGINED_DEVICE = new ConcurrentHashMap<>();
+    private static final String DEVICE_DATA_FILE_PATH = FileManager.getDeviceConfigPath() + Constant.DEVICE_DATA_FILE;
+
+    @PostConstruct
+    private void init() {
+        ObjectInputStream objectInputStream = null;
+        try {
+            File file = new File(DEVICE_DATA_FILE_PATH);
+            if (!file.exists()) {
```

Comment:
Why no logging before returning?

Reply:
We will add a backlog to enhance the logging

Reply:
Yes，we‘d better add log here and we will fix it.

Reply:
We have added the backlog to enhance the logging and error handling.
https://github.com/opensds/nbp/issues/250